### PR TITLE
tracker manager and http clients are no longer singletons

### PR DESCRIPTION
**Pull request by matanporeh**
now they are privately used via the BringgTrackingClient class
also updated tests to reflect the change

### CODE REVIEW

File: sources/model/managars/GGHTTPClientManager.m [hunk 1]
```diff
@@ -281,6 +266,16 @@ - (void)useCustomer:(GGCustomer * _Nullable)customer{
     self.customer = customer;
 }
```
Comment (orxelm) [inline sources/model/managars/GGHTTPClientManager.m]: Why did you duplicated the same code here?

File: sources/model/managars/GGTrackerManager.m [hunk 2]
```diff
@@ -32,85 +32,50 @@
...
-    
-    return sharedObject;
-}
-
-
--(nonnull instancetype)initTacker{
-    if (self = [super init]) {
-        self.logsEnabled = NO;
-    }
+        // set the customer token and developer token
```
Comment (orxelm) [inline sources/model/managars/GGTrackerManager.m]: else?

File: sources/model/managars/GGTrackerManager.m [hunk 3]
```diff
@@ -32,85 +32,50 @@
...
+        if (devToken) [self setDeveloperToken:devToken];
+        
+        // set the connection delegate
+        if (realTimeDelegate) [self setRealTimeDelegate:realTimeDelegate];
```
Comment (orxelm) [inline sources/model/managars/GGTrackerManager.m]: else?

File: sources/model/managars/GGHTTPClientManager.m (hunk 1 above)
Comment (matanporeh) [inline sources/model/managars/GGHTTPClientManager.m]: do you mean the `setLastEventDate:`?
there is one in the ggrealtimemonitor (realtime)
and there is one for the httpManager
its not a duplicate as the tracker manager wants to know, regardless of source when any kind of event was received

File: sources/model/managars/GGTrackerManager.m (hunk 2 above)
Comment (matanporeh) [inline sources/model/managars/GGTrackerManager.m]: what do you mean? if i get nil in dev token i don't do any thing with it

File: sources/model/managars/GGTrackerManager.m (hunk 3 above)
Comment (matanporeh) [inline sources/model/managars/GGTrackerManager.m]: what do you mean? if i get nil in the realtime delegate  i don't do any thing with it

**Pull request merged**